rfcs/0000-universal-asset-notation/0000-universal-asset-notation.md

```diff
@@ -0,0 +1,118 @@
+---
+Number: ""
+Category: Standards Track
+Status: Proposal
+Author: Jan Xie, Jordan Mack, Wenchao Hu
+Organization: Nervos Foundation, Cryptape
+Created: 2022-04-22
+---
+
+# Universal Asset Notation
+
+## Abstract
+
+This document specifies a universal naming method for assets flowing on Nervos Network, including assets on CKB and assets on layer 2 networks such as Godwoken and Axon.
+
+## Motivation
+
+The interoperation between Nervos Network and other crypto or traditional economy networks brings new assets to Nervos Network. The boom of assets create difficulties for developers and users - for example, different assets on different networks may accidentally use the same name, or the same asset may arrive Nervos Network through different cross-chain bridges/routes and eventually appears as different mapping assets. A proper asset notation helps incorporate assets origin from multiple networks into one network without causing any confusion, thus lowers the operational risk and facilitates developer and user experience.
+
+## Definition
+
+The Universal Asset Notation (UAN) consists of asset notation and route notation, which in turn are build upon asset symbol, chain symbol and bridge symbol.
+
+UAN Definition (in [BNF](https://en.wikipedia.org/wiki/Backus%E2%80%93Naur_form)):
+
+```
+<UAN> ::= <asset> | <asset> "|" <route>
+<asset> ::= <asset-symbol> "." <chain-symbol>
+<route> ::= <path> | <path> "|" <route>
+<path> ::= <bridge-symbol> "." <chain-symbol>
+<asset-symbol> ::= <upper-char>
+<chain-symbol> ::= <lower-char>
+<bridge-symbol> ::= <lower-char>
+<upper-char> ::= <upper-case> | <digit>
+<lower-char> ::= <lower-case> | <digit>
+<upper-case> ::= "A" | "B" | "C" | "D" | "E" | "F" | "G" | "H" | "I" | "J" | "K" | "L" | "M" | "N" | "O" | "P" | "Q" | "R" | "S" | "T" | "U" | "V" | "W" | "X" | "Y" | "Z"
+<lower-case> ::= "a" | "b" | "c" | "d" | "e" | "f" | "g" | "h" | "i" | "j" | "k" | "l" | "m" | "n" | "o" | "p" | "q" | "r" | "s" | "t" | "u" | "v" | "w" | "x" | "y" | "z"
+<digit> ::= "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9"
+```
+
+- Characters `|` and `.` are chosen in order to avoid confliction with `-` widely used in LP token in DeFi applications.
+- `<asset>` is the first component of UAN. It is a pair of `<asset-symbol>` and `<chain-symbol>` connected by a dot `.`, e.g. `BTC.ckb` for bitcoin resides on the CKB chain, `CKB.gw` for CKByte resides on the Godwoken chain.
+- `<route>` is the route through which the asset come to Nervos Network, it's the second component of UAN. `<route>` is only required when the asset's chain of residence is different from its chain of issuance. `<route>` may consist of one or many `<path>`. If it's a multi-paths route, all paths are connected by `|`.
+- `<path>` is a pair of bridge symbol and chain symbol, connected by a single dot `.`.
+- `<asset-symbol>` is a upper-case abbreviation of the asset, e.g. `BTC` for bitcoin, `CKB` for CKByte.
```

Comment:
The asset symbol should not be case restricted, since some ERC20 do have a lowercase symbol, like `stETH` (staked ETH), and some wrapped tokens, like `wETH`, `wMANA`